examples/todo_app/src/todo_list.rs

```diff
@@ -0,0 +1,142 @@
+
+        include!("cxx-qt-lib/qvariant.h");
+        type QVariant = cxx_qt_lib::QVariant;
+
+        include!("cxx-qt-lib/qstring.h");
+        type QString = cxx_qt_lib::QString;
+
+        include!("cxx-qt-lib/qhash.h");
+        type QHash_i32_QByteArray = cxx_qt_lib::QHash<cxx_qt_lib::QHashPair_i32_QByteArray>;
+    }
+
+    #[qenum(TodoList)]
+    enum TodoRoles {
+        Todo,
+        Done,
+    }
+
+    unsafe extern "RustQt" {
+        #[qobject]
+        #[qml_element]
+        #[base = QAbstractListModel]
+        type TodoList = super::TodoListRust;
+
+        #[cxx_override]
+        #[rust_name = "row_count"]
+        fn rowCount(self: &TodoList, parent: &QModelIndex) -> i32;
+
+        #[cxx_override]
+        fn data(self: &TodoList, index: &QModelIndex, role: i32) -> QVariant;
+
+        #[cxx_override]
+        #[rust_name = "role_names"]
+        fn roleNames(self: &TodoList) -> QHash_i32_QByteArray;
+    }
+
+    unsafe extern "RustQt" {
+        #[qinvokable]
+        #[rust_name = "set_checked"]
+        fn setChecked(self: Pin<&mut TodoList>, row: i32, checked: bool);
+
+        #[inherit]
+        #[rust_name = "begin_reset_model"]
+        fn beginResetModel(self: Pin<&mut TodoList>);
+
+        #[inherit]
+        #[rust_name = "end_reset_model"]
+        fn endResetModel(self: Pin<&mut TodoList>);
+
+        #[qinvokable]
+        #[rust_name = "add_todo"]
+        fn addTodo(self: Pin<&mut TodoList>, todo: &QString);
+    }
+}
+
+pub struct TodoListRust {
+    todos: Vec<(bool, QString)>,
+}
+
+impl Default for TodoListRust {
+    fn default() -> Self {
+        Self {
+            todos: vec![
+                (true, "Build ToDo Example".into()),
+                (false, "Modify ToDo Example".into()),
+            ],
+        }
+    }
+}
+
+use cxx_qt::CxxQtType;
+use qobject::*;
+
+impl qobject::TodoList {
+    fn row_count(&self, _parent: &QModelIndex) -> i32 {
+        self.todos.len() as i32
+    }
+
+    fn data(&self, index: &QModelIndex, role: i32) -> QVariant {
+        let role = TodoRoles { repr: role };
+
+        if let Some((done, ref todo)) = self.todos.get(index.row() as usize) {
+            match role {
+                TodoRoles::Todo => {
+                    return todo.into();
+                }
+                TodoRoles::Done => {
+                    return done.into();
+                }
+                _ => {}
+            }
+        }
+        QVariant::default()
+    }
+
+    fn role_names(&self) -> QHash_i32_QByteArray {
+        let mut hash = QHash_i32_QByteArray::default();
+        hash.insert(TodoRoles::Todo.repr, "todo".into());
+        hash.insert(TodoRoles::Done.repr, "done".into());
+        hash
+    }
+
+    fn set_checked(mut self: Pin<&mut Self>, row: i32, checked: bool) {
+        if let Some((done, _todo)) = self.as_mut().rust_mut().todos.get_mut(row as usize) {
+            if *done != checked {
+                *done = checked;
+                self.sort();
```

Comment:
same here, the sort is what is actually causing the beginResetModel / dataChanged that you'd usually emit

Reply:
Done